eventmesh-sdk-java/src/test/java/org/apache/eventmesh/client/grpc/consumer/EventMeshGrpcConsumerTest.java

```diff
@@ -0,0 +1,155 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+import org.apache.eventmesh.client.grpc.config.EventMeshGrpcClientConfig;
+import org.apache.eventmesh.common.EventMeshMessage;
+import org.apache.eventmesh.common.protocol.SubscriptionItem;
+import org.apache.eventmesh.common.protocol.SubscriptionMode;
+import org.apache.eventmesh.common.protocol.SubscriptionType;
+import org.apache.eventmesh.common.protocol.grpc.protos.ConsumerServiceGrpc.ConsumerServiceBlockingStub;
+import org.apache.eventmesh.common.protocol.grpc.protos.ConsumerServiceGrpc.ConsumerServiceStub;
+import org.apache.eventmesh.common.protocol.grpc.protos.HeartbeatServiceGrpc.HeartbeatServiceBlockingStub;
+import org.apache.eventmesh.common.protocol.grpc.protos.Response;
+import org.apache.eventmesh.common.protocol.grpc.protos.SimpleMessage;
+import org.apache.eventmesh.common.protocol.grpc.protos.Subscription;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.internal.util.reflection.FieldSetter;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import io.grpc.stub.StreamObserver;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ConsumerServiceBlockingStub.class, ConsumerServiceStub.class, HeartbeatServiceBlockingStub.class})
+@PowerMockIgnore({"javax.management.*", "com.sun.org.apache.xerces.*", "javax.xml.*", "org.xml.*", "org.w3c.*"})
+public class EventMeshGrpcConsumerTest {
+
+    @Mock
+    private ConsumerServiceBlockingStub consumerClient;
+    @Mock
+    private ConsumerServiceStub consumerAsyncClient;
+    @Mock
+    private HeartbeatServiceBlockingStub heartbeatClient;
+    private EventMeshGrpcConsumer eventMeshGrpcConsumer;
+
+    @Before
+    public void setUp() throws Exception {
+        eventMeshGrpcConsumer = new EventMeshGrpcConsumer(EventMeshGrpcClientConfig.builder().build());
+        eventMeshGrpcConsumer.init();
+        FieldSetter.setField(eventMeshGrpcConsumer, eventMeshGrpcConsumer.getClass().getDeclaredField("consumerClient"),
+            consumerClient);
+        FieldSetter.setField(eventMeshGrpcConsumer,
+            eventMeshGrpcConsumer.getClass().getDeclaredField("consumerAsyncClient"), consumerAsyncClient);
+        FieldSetter.setField(eventMeshGrpcConsumer,
+            eventMeshGrpcConsumer.getClass().getDeclaredField("heartbeatClient"), heartbeatClient);
+        when(consumerClient.subscribe(any())).thenReturn(Response.getDefaultInstance());
```

Comment:
Use reflect to create test object is not a good practice, When someone changes the field name, he cannot find out the problem in compile.
You can try to change the scope of field in `consumerClient`. The same to other test file.

Reply:
Yes, if someone change the field name, error will only be thrown when the test case is run. Change the field scope to `protected` will resolve the problem. I have another idea, what about we create a setter method with protected modification and `VisibleForTest` comment. Like this:

```java
private ConsumerServiceBlockingStub consumerClient;

/**
*VisibleForTest
**/
protected void setConsumerClient(ConsumerServiceBlockingStub consumerClient) {
    this.consumerClient = consumerClient;
}
```

Reply:
I would like to change the field to default or protected, no need to add setter method, in fact, we shouldn't add a method to modify this field.

Reply:
OK, I try to fix it.